File: Bin.py
class Bin:
  def __init__(self, id, lowerLimit, upperLimit):
    self.id = id
    self.lowerLimit = lowerLimit
    self.upperLimit = upperLimit
    self.numberOfBalls = 0
    self.currentValue = 0
  
def UpdateValue(self):
  self.currentValue = self.numberOfBalls*(self.numberOfBalls+1)/2
  return self.currentValue

def SetBalls(self, targetNumBalls):
  self.numberOfBalls = targetNumBalls
  return UpdateValue(self)
  

def IsValidToPut(self):
  if(self.numberOfBalls+1<=self.upperLimit):
    return True
  else:
    return False
def IsValidToTake(self):
  if(self.numberOfBalls-1>=self.lowerLimit):
    return True
  else:
    return False


  

File: Bin.java
public class Bin {
    private int id;
    private int lowerLimit;
    private int upperLimit;
    private int numberOfBalls;
    private double currentValue;

    public Bin(int id, int lowerLimit, int upperLimit,int numberOfBalls) {
        this.id = id;
        this.lowerLimit = lowerLimit;
        this.upperLimit = upperLimit;
        this.numberOfBalls = numberOfBalls;
        this.currentValue = 0;
    }
    
    public Bin(int id, int lowerLimit, int upperLimit) {
        this.id = id;
        this.lowerLimit = lowerLimit;
        this.upperLimit = upperLimit;
        this.numberOfBalls = 0;
        this.currentValue = 0;
    }


    public double updateValue() {
        this.currentValue = this.numberOfBalls * (this.numberOfBalls + 1) / 2.0;
        return this.currentValue;
    }

    public double setBalls(int targetNumBalls) {
        this.numberOfBalls = targetNumBalls;
        return updateValue();
    }

    public boolean isValidToPut() {
        return this.numberOfBalls + 1 <= this.upperLimit;
    }

    public boolean isValidToTake() {
        return this.numberOfBalls - 1 >= this.lowerLimit;
    }

    // Getters and Setters (opcional, dependendo da necessidade)
    public int getId() {
        return id;
    }

    public int getLowerLimit() {
        return lowerLimit;
    }

    public int getUpperLimit() {
        return upperLimit;
    }

    public int getNumberOfBalls() {
        return numberOfBalls;
    }

    public double getCurrentValue() {
        return currentValue;
    }
}
